### PR TITLE
163383144 Implement Social Share articles

### PR DESCRIPTION
**Pull request by missvicki**
**Description**

The ability to share/tweet/post an article on social media and through email increases the reach of the article and, hence, the popularity of the platform.

**Type of change**

- New feature

**How has this been tested**

- Unit tests testing individual functionalities

**What PR does?**

- modify tests to incorporate sharing articles
- render sharable links
- use redux to modify state

**Relevant Pivotal Tracker Stories** 

[#163383144](https://www.pivotaltracker.com/story/show/163383144)

### CODE REVIEW

File: src/store/reducers/article.js [hunk 1]
```diff
@@ -5,81 +5,87 @@ const initiaState = {
...
+    case actionTypes.GET_ARTICLE_SUCCESS:
+        return {
+            ...state,
+            loading: false,
+            article: action.data.articles,
+            error: null,
+            averageRating: action.data.articles.average_ratings,
+            facebook:action.data.articles.share_links.facebook,
+            twitter:action.data.articles.share_links.twitter,
+            email:action.data.articles.share_links.email
```
Comment (fahdjamy) [inline src/store/reducers/article.js]: @missvicki I feel like this is a little bit not lean. I mean could I get a clear reason to why you decided to add twitter, email and Facebook states in the reducer. I mean the article state that was initially set was containing all this and all you had to do is just desctructre it and get all the information you needed.

File: src/store/reducers/article.js [hunk 2]
```diff
@@ -5,81 +5,87 @@ const initiaState = {
     error: null,
     loading: false,
     message: "",
-    averageRating: null
+    averageRating: null,
+    facebook: null,
+    email: null,
+    twitter: null
```
Comment (fahdjamy) [inline src/store/reducers/article.js]: I feel like all this information is already contained in the article variable here in this article reducer state so this is kind of some redundancy. Could I get a cleary reason though to why you decided to do this?

**Review by fahdjamy — COMMENTED** [review]
Good work you pulled off here @missvicki.
I feel like you could have written a few lines of code during this implementation though, I have left a few reviews to what you might want to look into.
Thanks for the good work though.

**Review by karuhanga — CHANGES_REQUESTED** [review]
The share buttons work beautifully on this. 👍 
However;
- Is there any reason we restrict sharing to only authenticated users?
- Could the colors for the share buttons reflect the platforms they point to?
- What do you think about possibly centering the icons?
image
image
this has been worked on

**Comment by missvicki** [comment]
> The share buttons work beautifully on this. 👍
> However;
> 
> * Is there any reason we restrict sharing to only authenticated users?
> * Could the colors for the share buttons reflect the platforms they point to?
> * What do you think about possibly centering the icons?
>   image
>   image

Screenshot 2019-03-27 at 17 11 06
this should explain why sharing is restricted

**Comment by codjoero** [comment]
@missvicki Good work on the functionality. On the UI side though, the icons are quite huge even compared to the rating stars you previously implemented.

File: src/store/reducers/article.js (hunk 1 above)
Comment (missvicki) [reply src/store/reducers/article.js]: Thank you for paying attention to this. I am working on this.

File: src/store/reducers/article.js (hunk 2 above)
Comment (missvicki) [reply src/store/reducers/article.js]: working on this as well

**Review by karuhanga — APPROVED** [review]

File: src/components/UI/Article/Article.jsx [hunk 3]
```diff
@@ -16,46 +16,60 @@ const article = (props) => {
...
             <Container textAlign='justified'>
-                <Image src= { image ? image : defautImage} />
+                <Image src={image ? image : defautImage} />
                 <div className="ArticleBody">
                     {renderedBody}
                 </div>
             </Container>
+            {isLoggedIn ? (
+                <Container textAlign="center">
+                    <a href={facebook} style={{ color: "#4267b2" }}><Icon name="facebook f" style={{ fontSize: "24px" }} /></a>
```
Comment (karuhanga) [inline src/components/UI/Article/Article.jsx]: Is there any reason we are the `styling` is here and not in the `css` files?

**Review by karuhanga — CHANGES_REQUESTED** [review]
Good work on the changes @missvicki. Especially with getting rid of the redundancy. One more thing;

File: src/components/UI/Article/Article.jsx (hunk 3 above)
Comment (missvicki) [reply src/components/UI/Article/Article.jsx]: Just worked on this

**Review by karuhanga — APPROVED** [review]

**Review by codjoero — APPROVED** [review]
LGTM

**Review by fahdjamy — APPROVED** [review]
Great job implementing all the changes @missvicki.

**Pull request merged**